src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -144,13 +144,18 @@ public static bool InStandbyMode
                 {
                     return _standbyMode.Value;
                 }
-                if (ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.AzureWebsitePlaceholderMode) == "1")
+
+                if (ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.AzureWebsitePlaceholderMode) == "0" 
+                    && ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.AzureWebsiteContainerReady) == "1"
+                    && ScriptSettingsManager.Instance.GetSetting(EnvironmentSettingNames.AzureWebsiteConfigurationReady) == "1")
```

Comment:
Yikes, we were never checking ContainerReady in v1?!

And what is ConfigurationReady? I don't see this used anywhere in v2 either (and I don't think I'd heard of it). I'm assuming we need to do this in v2 as well?

Reply:
Updated description with details on when these environment variables get set. ConfigurationReady is V1 only setting. We do have a check for these settings here: 
https://github.com/Azure/azure-functions-host/blob/9b968f8605e3bc13c28435a8ce1a4b5d777c7b74/src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs#L180-L181

but if WEBSITE_PLACEHOLDER_MODE is set earlier and if there is a delay in setting other environment variables, in this case there was  large zip file download, the we completely skip specialization.

Testing this change on private stamp now.

Reply:
Ahh, okay, that's making more sense. Thanks! As long as your testing looks good, go ahead and merge.